docs/source/common/optimizers.rst

```diff
@@ -137,8 +135,12 @@ Here is an example on how to use it:
 
 Automatic optimization
 ======================
-With Lightning most users don't have to think about when to call .backward(), .step(), .zero_grad(), since
-Lightning automates that for you.
+With Lightning most users don't have to think about when to call ``.zero_grad()``, ``.backward()`` and ``.step()``
+since Lightning automates that for you.
+
+.. warning::
+   Before 1.2.1, ``.zero_grad()`` was called after ``.backward()`` and ``.step()`` internally.
+   From 1.2.1, Lightning calls ``.zero_grad()`` before ``.backward()``.
```

Comment:
isn't it better to move this into 1.3? @carmocca 
also 1.2.1 is already out, so this message must be updated anyway

Reply:
@awaelchli I don't think so, since this is actually a bugfix to finally support lbfgs

Reply:
@awaelchli Updated to 1.2.2, but I'll update it to 1.3 if needed.

Reply:
> this is actually a bugfix to finally support lbfgs

And (some?) GANs